conda_package_streaming/extract.py

```diff
@@ -27,24 +27,22 @@ def extract_stream(
     For ``.conda`` will need to be called twice (for info and pkg components);
     for ``.tar.bz2`` every member is extracted.
     """
-    dest_dir = str(dest_dir)
-    # we don't extract to cwd; only used to check that tar member does not
-    # escape its target directory:
-    cwd = os.getcwd()
+    dest_dir = os.path.realpath(dest_dir)
+
+    def is_within_dest_dir(name):
+        abs_target = os.path.realpath(os.path.join(dest_dir, name))
+        prefix = os.path.commonpath((dest_dir, abs_target))
```

Comment:
this should be the common entire path unlike the older (< python 3.5 or 3.6) commonprefix